ls1intum/Artemis
# `General`: Use a dropdown menu for filtering exercises in the students exercise list

### Checklist
#### General


#### Client

### Motivation and Context



Previously, all filters on the exercise list were raw labeled checkboxes in the top row. On small screen sizes, this may overlap other items in the top row, and as we're adding more filters, things become cluttered. Therefore, this PR moves all the filters to a dropdown menu.

### Description


Moved all filter checkboxes to a dropdown menu. The button will show how many filters are active and change the color if at least one is active to make it obvious that filters are active.

Since I'm already working on it, this PR closes #4326 as well.

### Steps for Testing

Prerequisites:
- 1 Student
- A few exercises that can be filtered. Have an optional exercise, an overdue one, one with full points ...

1. Log in to Artemis
2. Open a course as a student
3. Use the new dropdown to filter the list.
4. Verify that the list of shown exercises changes as before.
5. Verify that the button changes the background color, and that the number in it shows the appropriate number of active filters.

### Review Progress





#### Code Review
#### Manual Tests

### Test Coverage






### Screenshots



Before:
![grafik](https://user-images.githubusercontent.com/23171488/143961037-44b485bd-e50e-4d4a-88e8-30d2f44fba9f.png)

After:
![grafik](https://user-images.githubusercontent.com/23171488/143961057-e2520744-4cc6-4567-b9e3-78df8b311517.png)


https://user-images.githubusercontent.com/23171488/144122715-709d9232-a740-4ecf-970b-600a1d67131e.mov
